### PR TITLE
remove all mocking of web3.eth from walletService test

### PR DESCRIPTION
**Pull request by cellog**
# Description

This test removes meddling with the internal details of `web3.eth` using `jest.fn()` from `walletService.test.js`. However, it leaves in place a mock provider, which will need to be updated with all upgrades.

# Issues

Refs #2782

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-js/src/__tests__/walletService.test.js [hunk 1]
```diff
@@ -119,13 +117,7 @@ describe('WalletService', () => {
...
           const newAccount = '0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1'
 
-          walletService.web3.eth.accounts.create = jest.fn(() => {
-            return Promise.resolve({
-              address: newAccount,
-            })
-          })
-
-          nock.accountsAndYield([])
+          nock.accountsAndYield([newAccount])
```
Comment (julien51) [inline unlock-js/src/__tests__/walletService.test.js]: This is not right. The test actually verifies that when no account is available on the node, one is created by web3... So we cannot have a mock actually return an account.

File: unlock-js/src/__tests__/walletService.test.js [hunk 2]
```diff
@@ -152,110 +144,127 @@ describe('WalletService', () => {
...
+            gasUsed: '0x16e360',
+            cumulativeGasUsed: '0x16e360',
+            logs: [],
+          })
+        })
 
-      it('should handle cases where the transaction is sent via a provider', () => {
-        expect.assertions(1)
+        it('should handle cases where the transaction is sent via a provider', async () => {
+          expect.assertions(0)
```
Comment (julien51) [inline unlock-js/src/__tests__/walletService.test.js]: hu, wat? This is probabluy not a good test if there are no assertions ;p

File: unlock-js/src/__tests__/walletService.test.js (hunk 1 above)
Comment (cellog) [reply unlock-js/src/__tests__/walletService.test.js]: ah that makes sense, I'll add comments for the next victim so we don't forget :)

File: unlock-js/src/__tests__/walletService.test.js (hunk 2 above)
Comment (cellog) [reply unlock-js/src/__tests__/walletService.test.js]: I'll read up on nock, see if we can get something to assert against from it

**Comment by cellog** [comment]
ok @julien51 PTAL. The solution I came up with was to make a test where the nock calls are explicit, and that we then have a helper that verifies we have used all nocks, and not had any mis-matched nocks in the event that a test does not fail but the nocks mismatch. I also fixed that test so that it calls the real account creation method, and verifies that we got an account hash back.

File: unlock-js/src/__tests__/walletService.test.js [hunk 3]
```diff
@@ -152,110 +146,155 @@ describe('WalletService', () => {
...
-            expect(input).toEqual(data)
-            expect(type).toEqual('type')
-            expect(status).toEqual('submitted')
-            done()
-          }
-        )
+        // in the place of assertions, we explicitly verify that all of the JSON-RPC
+        // calls were completed in the exact order specified above, and that all
+        // of them were called, and nothing else
+        nock.ensureAllNocksUsed()
```
Comment (julien51) [inline unlock-js/src/__tests__/walletService.test.js]: Works for me.

**Pull request merged**